src/state/slices/actionSlice/types.ts

```diff
@@ -0,0 +1,125 @@
+import type { AccountId, AssetId } from '@shapeshiftoss/caip'
+import type { Asset, OrderId, QuoteId } from '@shapeshiftoss/types'
+
+import type { PriceDirection } from '../limitOrderInputSlice/constants'
+
+export enum ActionCenterType {
+  Deposit = 'Deposit',
+  Claim = 'Claim',
+  Swap = 'Swap',
+  LimitOrder = 'LimitOrder',
+}
+
+export enum ActionStatus {
+  Pending = 'Pending',
+  Complete = 'Complete',
+  Failed = 'Failed',
+  ClaimAvailable = 'ClaimAvailable',
+  Claimed = 'Claimed',
+  Open = 'Open',
+  Expired = 'Expired',
+  Cancelled = 'Cancelled',
+}
+
+export type ActionId = string
+
+type ActionSwapMetadata = {
+  swapId: string
+}
+
+export type ActionLimitPrice = {
+  [PriceDirection.BuyAssetDenomination]: string
+  [PriceDirection.SellAssetDenomination]: string
+}
+
+type ActionLimitOrderMetadata = {
+  quoteId: QuoteId
+  limitOrderId?: OrderId
+  sellAmountCryptoBaseUnit: string
+  buyAmountCryptoBaseUnit: string
+  sellAsset: Asset
+  buyAsset: Asset
+  limitPrice: ActionLimitPrice
+  expires: number | undefined
+  executedBuyAmountCryptoBaseUnit?: string
+  executedSellAmountCryptoBaseUnit?: string
+  filledDecimalPercentage?: string
+}
+
+export type SwapAction = {
+  id: ActionId
+  type: ActionCenterType.Swap
+  status: ActionStatus
+  createdAt: number
+  updatedAt: number
+  title: string
+  assetIds: AssetId[]
```

Comment:
q: can't we rely on the swap itself for that? i.e keep the action as atomic as possible and then join with swaps?